aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -66,23 +68,20 @@
     private final AppSync appSync;
     private final ModelProvider modelProvider;
     private final Merger merger;
+    private final DataStoreConfigurationProvider dataStoreConfigurationProvider;
     private final CompositeDisposable ongoingOperationsDisposable;
     private final long adjustedTimeoutSeconds;
     private ReplaySubject<SubscriptionEvent<? extends Model>> buffer;
 
     /**
      * Constructs a new SubscriptionProcessor.
-     * @param appSync An App Sync endpoint from which to receive subscription events
-     * @param modelProvider The processor will subscribe to changes for these types of models
-     * @param merger A merger, to apply data back into local storage
+     * @param builder A SubscriptionProcessor Builder.
      */
-    SubscriptionProcessor(
-            @NonNull AppSync appSync,
-            @NonNull ModelProvider modelProvider,
-            @NonNull Merger merger) {
-        this.appSync = Objects.requireNonNull(appSync);
-        this.modelProvider = Objects.requireNonNull(modelProvider);
-        this.merger = Objects.requireNonNull(merger);
+    private SubscriptionProcessor(Builder builder) {
+        this.appSync = Objects.requireNonNull(builder.appSync);
```

Comment:
As sort of a philosophical point, I'd argue the non-null checks should be in the `build()`, not in the `private` constructor. It is to say "We won't even attempt construction of the object, if user-provided options `foo`, `bar`, are null."

Reply:
That philosophy makes sense!    I'll remove these here in the constructor.  I actually don't think I need to add them to `build()`, because the step methods call `Objects.requireNonNull(...)` already, and the step builder makes all arguments required.